serverlessgo/main.go

```diff
@@ -133,27 +138,59 @@ func (i *instance) receiveRequest(r *request) {
 }
 
 func (i *instance) terminate() {
-	i.isTerminated = true
+	i.terminateTime = godes.GetSystemTime()
```

Comment:
Isso vai fazer fazer com o que o uptime seja calculado errado. Isto pois o tryScaleDown não é chamado exatamente na hora certa. 

Aqui, como é simulação, você precisa fazer uns ajustes. Se for scale down, o tempo tem que ser getLastWorkTime + idlesnessDeadline. Se a simulação terminou, pode ser GetSystemTime. Entende?

Reply:
Concordo contigo. Haverá uma imprecisão para fazer o scale down, uma vez que tryScaleDown só acontece após o LB redirecionar uma requisição para uma instância. Como consequência, pode acontecer de instâncias serem terminadas após terem passado mais tempo ocioso que o definido no idlenessDeadline. Contudo, por que isso seria um problema?

Reply:
Por que isso vai aumentar o uptime das instâncias, o aumentará o custo de forma errônea. Ou seja, estaremos calculando métricas de forma errada.